src/gui/properties/speedplotview.cpp

```diff
@@ -33,15 +33,61 @@
 #include "base/global.h"
 #include "base/utils/misc.h"
 
+namespace
+{
+    enum PeriodInSeconds
+    {
+        MIN1_SEC = 60,
+        MIN5_SEC = 5 * 60,
+        MIN30_SEC = 30 * 60,
+        HOUR6_SEC = 6 * 60 * 60
+    };
+
+    const int MIN5_BUF_SIZE = 5 * 60;
+    const int MIN30_BUF_SIZE = 5 * 60;
+    const int HOUR6_BUF_SIZE = 5 * 60;
+    const int DIVIDER_30MIN = MIN30_SEC / MIN30_BUF_SIZE;
+    const int DIVIDER_6HOUR = HOUR6_SEC / HOUR6_BUF_SIZE;
+};
+
+SpeedPlotView::Averager::Averager(int divider, boost::circular_buffer<PointData> &sink)
+    : m_divider(divider)
+    , m_sink(sink)
+    , m_counter(0)
+    , m_accumulator({})
+{
+}
+
+void SpeedPlotView::Averager::push(const PointData &pointData)
+{
+    m_accumulator.x += pointData.x;
+    for (int id = UP; id < NB_GRAPHS; ++id)
+        m_accumulator.y[id] += pointData.y[id];
```

Comment:
Maybe it is better to add `PointData::operator+=()`

Reply:
also there is possible integer overflow here.

Reply:
I think it not better. Because this new method:
1.  will be used only in 1 place;
2.  too simple to hide something and make programmers' live easier;
3.  adds function call overhead.

Reply:
Old code was vulnerable to overflow too. As soon as download speeds get to 29826161 byte/s (30Mbyte/s approximately) here (in new code) will be overflow. Definitely it is time to do something. I think the best way is to change type of `PointData.y` to double.

Reply:
> I think the best way is to change type of PointData.y to double.

Why a floating point type? what about qint64 or quint64?

Reply:
Also AFAIK the overflow only occurs at the accumulator, maybe just use quint64 for its type.

Reply:
Switch from int to double is really bad idea for "increase performance" patch.

Reply:
I have seen posts on net that some people already have 10Gbit/s connection on their premise. So not only the accumulator needs to be enlarged, but data storage too. 
The choice is of two options: double or qint64. 
Modern processors are pretty fast with float arithmetic, thanks to billions of transistors on a dye. So I will do measurements and then choose. Please, wait.

Reply:
Hmm. it look a bit surprising, but double wins! After filtering out excesses due background processes, I got such results:
With double - 8282 ticks.
with qint64 - 8744 ticks. 
For measuring code look into corresponding commit.

Reply:
I compile to 32bit app.

Reply:
64 bit application should calculate 64 bit integers more efficient, isn't it?

Reply:
@dzmat, you can compare int vs int64 using 32bit app.

Reply:
Finally managed to build 64bit version and make tests. 
Double on 64bit use 5223 ticks.
qint64  on 64bit use 6353 ticks.
Double wins here too for my surprise.

Reply:
Made [branch](https://github.com/dzmat/qBittorrent/tree/averager_instrumented) with measure code if somebody want to play with it. Removed code from here

Reply:
@dzmat 
qbt also runs on other platforms: embedded systems, ARM, MIPS... It is always a good idea to avoid floating-point types if not really required.

Reply:
Slowing going offtopic :\ 
>But is there ARM/MIPS computers powerful enough to run qt5 gui on a GUI-enabled OS and in the same time having such a bad FPU abilities?

I cannot say for sure, however a bad FPU is still a FPU, still usable, just not as fast as general-purpose registers. And remember there are ARMs that runs Win10.

Back to the topic.
I would adopt this algorithm instead: https://stackoverflow.com/a/1934266 
and use `quint64` for input & output, also change to integer division, will lose a bit precision but it is fine IMO.

Reply:
Very nice algorithm! 3 operations per iteration but elided overflow without using extended range types. 
Well I'll go to quint64 just to stop discussion and make this PR accepted.

Reply:
I don't take this algorithm because is is an overkill. For us the quint64 type is enough.

Reply:
OK, then at least put a comment documenting the possible overflow and why quint64 is enough.